lib/waylon/jenkins/job/rest.rb

```diff
@@ -86,7 +92,23 @@ def disabled?
         end
 
         def last_build_timestamp
-          @client.job.get_build_details(@name, last_build_num)['timestamp']
+          @last_build_timestamp ||= @client.job.get_build_details(@name, last_build_num)['timestamp']
+        end
+
+        def since_last_build
+          # Figure out the number of seconds between the last_build_timestamp and now.
+          # using to_i uses UTC based time so we need to facter in the local timezone to get the total timespan
+          t = (Time.now.to_i - Time.at(last_build_timestamp / 1000).to_i) - Time.now.utc_offset.to_i
+
+          # Convert numer of seconds into something nicer to view
+          dd, hh, mm = [t/86400, t/3600%24, t/60%60].map! { |x| x.floor }
```

Comment:
It seems like this date string formatting code could be extracted and used in both places.